Library/Homebrew/utils/pypi.rb

```diff
@@ -167,30 +170,34 @@ def update_python_resources!(formula, version: nil, package_name: nil, extra_pac
       end
     end
 
+    input_packages = []
     if main_package.blank?
-      return if ignore_non_pypi_packages
-
-      odie <<~EOS
-        Could not infer PyPI package name from URL:
-          #{Formatter.url(formula.stable.url)}
-      EOS
-    end
-
-    unless main_package.valid_pypi_package?
-      return if ignore_non_pypi_packages
-
-      odie "\"#{main_package}\" is not available on PyPI."
+      if ignore_non_pypi_packages
+        return if extra_packages.blank?
+      else
+        odie <<~EOS
+          Could not infer PyPI package name from URL:
+            #{Formatter.url(formula.stable.url)}
+        EOS
+      end
```

Comment:
```suggestion
      if ignore_non_pypi_packages
        return if extra_packages.blank?
      end
      
      odie <<~EOS
        Could not infer PyPI package name from URL:
          #{Formatter.url(formula.stable.url)}
      EOS
```

Nice to remove/reduce this identation when possible on `return` or `odie`. Can do the same below too.

Reply:
Want to allow continuing when `ignore_non_pypi_packages && extra_packages.present?`, i.e.:
- `ignore_non_pypi_packages && extra_packages.blank?` --> return
- `ignore_non_pypi_packages && extra_packages.present?` --> only update resources for extra_packages
- `!ignore_non_pypi_packages` --> odie

Reply:
Ok, I see.